src/lib/storage/column_accessor.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <type_traits>
+
+#include "resolve_type.hpp"
+#include "storage/reference_column.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+/**
+ * This is the base class for all ColumnAccessor types.
+ * It provides the common interface to access individual values of a column.
+ */
+template <typename T>
+class BaseColumnAccessor {
```

Comment:
Even though it's minimal, this violates the one-class-per-compilation-unit rule
please transfer this to its own header.

Reply:
added `base_column_accessor.hpp`